src/app/shared/filter-by-property/filter-by-property.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterByPropertyPipe implements PipeTransform {
+    transform<T>(
+        items: T[] | undefined | null,
+        propertyName: keyof T,
+        searchValue: unknown,
+    ): T[] | undefined | null {
+        if (!Array.isArray(items) || !propertyName || typeof searchValue === 'undefined') {
+            return items;
+        }
+
+        return items.filter(item => {
+            if (typeof item[propertyName] === 'string') {
+                return (item[propertyName] as string)
```

Comment:
Подкину идею, как можно было бы уйти от `as` у `item[propertyName]`
```suggestion
            const itemValue = item[propertyName];
            
            if (typeof itemValue === 'string') {
                return itemValue
```

Кажется и мелочь, но на мой взгляд код становится причесанней и чище)

Reply:
поправило, действительно стало получше выглядеть, спасибо!